packages/flutter/lib/src/services/text_input.dart

```diff
@@ -590,12 +589,12 @@ class TextInputConfiguration {
       'inputAction': inputAction.toString(),
       'textCapitalization': textCapitalization.toString(),
       'keyboardAppearance': keyboardAppearance.toString(),
-      if (autofillConfiguration != null) 'autofill': autofillConfiguration.toJson(),
+      if (autofillConfiguration != null) 'autofill': autofillConfiguration!.toJson(),
     };
   }
 }
 
-TextAffinity _toTextAffinity(String affinity) {
+TextAffinity? _toTextAffinity(String affinity) {
```

Comment:
This is odd. Returning null from here is always a bug, it means we missed a case in the switch. This should really be non-nullable (unless null is a legal value for affinity).

Reply:
Changed